File: examples/performance/primes/Primes.cs
namespace FHNW;

public class SimplePrimes
{
    private const int Size = 1_000_000;
    private static readonly int[] _primes = new int[Size];

    public static void Main()
    {
        _primes[0] = 2;
        _primes[1] = 3;

        var n = 5;
        var i = 2;
        var j = 1;
        var limit = Math.Pow(n, 0.5);

        while (i < Size)
        {
            var isPrime = true;

            for (j = 1; isPrime && _primes[j] < limit && j < i; j++)
            {
                if (n % _primes[j] == 0)
                {
                    isPrime = false;
                }
            }

            if (isPrime)
            {
                _primes[i] = n;
                i++;

		Console.WriteLine(n);
            }

            n += 2;
            limit = Math.Pow(n, 0.5);
        }
    }
}

File: examples/performance/primes/primes.php
<?php // php -d opcache.enable_cli=1 -d opcache.jit_buffer_size=32M ./primes.php
$known = array(2, 3);
list($n, $i, $j) = array(5, 2, 1);
$limit = pow($n, 0.5);

while ($i < 1000000) {
    $isPrime = true;

    for ($j = 1; $isPrime && $known[$j] < $limit && $j < $i; $j++) {
        if ($n % $known[$j] == 0) {
            $isPrime = false;
        }
    }

    if ($isPrime) {
        $known[$i] = $n;
        $i++;

        echo($n . PHP_EOL);
    }

    $n += 2;
    $limit = pow($n, 0.5);
}
